### PR TITLE
Added GCP SQL plugins for database flags.

### PR DESCRIPTION
**Pull request by umermehmood**

### CODE REVIEW

File: plugins/google/sql/postgresqlLogMinDuration.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                                                                        flag.value && flag.value == '-1');
+
+                        if (found) {
+                            helpers.addResult(results, 0, 
+                                'SQL instance have log_min_duration_statement flag enabled', region, sqlInstance.name);
+                        } else {
+                            helpers.addResult(results, 2,
+                                'SQL instance does not have log_min_duration_statement flag enabled', region, sqlInstance.name);
+                        }
+                } else if (sqlInstance.instanceType == "READ_REPLICA_INSTANCE") {
```
Comment (dipsubha06) [inline plugins/google/sql/postgresqlLogMinDuration.js:55]: Is this else if needed. if so can we have some handling in that case

File: plugins/google/sql/postgresqlLogMinDuration.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+                if (sqlInstance.instanceType != "READ_REPLICA_INSTANCE" &&
+                    sqlInstance.settings &&
+                    sqlInstance.settings.databaseFlags &&
+                    sqlInstance.settings.databaseFlags.length) {
+                        let found = sqlInstance.settings.databaseFlags.find(flag => flag.name && flag.name == 'log_min_duration_statement' &&
+                                                                        flag.value && flag.value == '-1');
+
+                        if (found) {
+                            helpers.addResult(results, 0,
```
Comment (dipsubha06) [inline plugins/google/sql/postgresqlLogMinDuration.js:49]: from the description of the plugin it is said that the flag should not be enabled. then this should be a failing case if the flag is enabled,right?

**Review by dipsubha06 — CHANGES_REQUESTED** [review]

File: plugins/google/sql/postgresqlLogTempFiles.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
+var async = require('async');
+var helpers = require('../../../helpers/google');
+
+module.exports = {
+    title: 'PostgreSQL Log Temp Files Flag Enabled',
```
Comment (AkhtarAmir) [inline plugins/google/sql/postgresqlLogTempFiles.js:5]:
```suggestion
    title: 'PostgreSQL Log Temp Files',
```

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/google/sql/postgresqlLogMinDuration.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+var async = require('async');
+var helpers = require('../../../helpers/google');
+
+module.exports = {
+    title: 'PostgreSQL Log Min Duration Statement',
+    category: 'SQL',
+    description: 'Ensures SQL instances for PostgreSQL type have log min duration statement flag disabled.',
+    more_info: 'SQL instance for PostgreSQL databases provides log_min_duration_statement flag. It is used to log the duration of every completed statement. We should always disable it as there can be sensitive information as well that should not be recorded in the logs.',
```
Comment (giorod3) [inline plugins/google/sql/postgresqlLogMinDuration.js:8]:
```suggestion
    more_info: 'SQL instance for PostgreSQL databases provides log_min_duration_statement flag. It is used to log the duration of every completed statement. This should always be disabled as there can be sensitive information as well that should not be recorded in the logs.',
```

File: plugins/google/sql/postgresqlLogMinDuration.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                        if (found) {
+                            helpers.addResult(results, 0, 
+                                'SQL instance does not have log_min_duration_statement flag enabled', region, sqlInstance.name);
+                        } else {
+                            helpers.addResult(results, 2,
+                                'SQL instance have log_min_duration_statement flag enabled', region, sqlInstance.name);
+                        }
+                } else if (sqlInstance.instanceType == "READ_REPLICA_INSTANCE") {
+                } else {
+                    helpers.addResult(results, 2,
```
Comment (giorod3) [inline plugins/google/sql/postgresqlLogMinDuration.js:57]: this should be a pass if there are no database flags

**Review by giorod3 — CHANGES_REQUESTED** [review]
two comments

**Review by giorod3 — APPROVED** [review]

**Pull request merged**